### ISSUE TITLE
Meaning of loss during training

### ISSUE THREAD

**Issue by getao** [OpenNMT/OpenNMT-tf#50]
Hello, I want to know if loss printed on the screen during training a seq2seq model is the same with the Perpleixty in OPENNMT-lua. I found the loss changes very significantly during training using OpenNMT-tf while the Perplexity in OpenNMT-lua (the same model with the same setting including learning rate and optimizer) decreases almost monotonously during training.

I listed a snippet of my training log here:

INFO:tensorflow:loss = 48.18139, step = 1001 (46.707 sec)
INFO:tensorflow:global_step/sec: 2.09447
INFO:tensorflow:words_per_sec/features:0: 3275.1
INFO:tensorflow:words_per_sec/labels:0: 3610.39
INFO:tensorflow:global_step/sec: 2.13073
INFO:tensorflow:words_per_sec/features:0: 2945.64
INFO:tensorflow:words_per_sec/labels:0: 3282.78
INFO:tensorflow:loss = 61.09418, step = 1101 (47.340 sec)
INFO:tensorflow:global_step/sec: 2.08646
INFO:tensorflow:words_per_sec/features:0: 3127.07
INFO:tensorflow:words_per_sec/labels:0: 3464.33
INFO:tensorflow:global_step/sec: 2.14416
INFO:tensorflow:words_per_sec/features:0: 3226.2
INFO:tensorflow:words_per_sec/labels:0: 3564.5
INFO:tensorflow:loss = 110.21961, step = 1201 (47.280 sec)
INFO:tensorflow:global_step/sec: 2.23839
INFO:tensorflow:words_per_sec/features:0: 3106.01
INFO:tensorflow:words_per_sec/labels:0: 3462.01
INFO:tensorflow:global_step/sec: 2.2783
INFO:tensorflow:words_per_sec/features:0: 3460.2
INFO:tensorflow:words_per_sec/labels:0: 3825.8
INFO:tensorflow:loss = 9.316487, step = 1301 (44.303 sec)

Thanks in advance.

**Comment by guillaumekln** [comment]
Hello,

* OpenNMT-lua reports a normalized loss that is also averaged since the previous print
* OpenNMT-tf reports the loss that is actually compute at the current step

See also #38.

**Comment by getao** [comment]
So, does the normalized loss mean that the loss normalized to the length of a sentence?
Thanks

**Comment by guillaumekln** [comment]
Normalized by the number of target tokens.

**Comment by getao** [comment]
Thank you. It exactly accounts for that loss in OPENNMT-tf is much larger than that in OPENNMT-lua.
